### PR TITLE
279_getquote resillient to invalid symbols

### PR DESCRIPTION
**Pull request by ethanbsmith**
getQuote would fail if the source did not recognize any of the supplied symbols. this was particularly problematic on large batches where a single failure would cause a stop and there was no way to diagnose which symbol had caused the problem

all getQuote paths now support ";" delimited inputs 
All getQuote paths now return NA rows for missing quotes

### CODE REVIEW

**Comment by ethanbsmith** [comment]
ready for review. could not add tests for tiingo and av as these require a key

File: R/getQuote.R [hunk 1]
```diff
@@ -326,6 +332,8 @@ getQuote.av <- function(Symbols, api.key, ...) {
          "Registration at https://api.tiingo.com/.", call. = FALSE)
   }
 
+  Symbols <- unlist(strsplit(Symbols,';'))
```
Comment (joshuaulrich) [inline R/getQuote.R]: It's good practice to only have one functional change per commit, so this should be in another commit.

File: R/getQuote.R [hunk 2]
```diff
@@ -369,11 +377,14 @@ getQuote.av <- function(Symbols, api.key, ...) {
   }
 
   colnames(r) <- gsub("(^[[:alpha:]])", "\\U\\1", colnames(r), perl = TRUE)
-  r[, "Trade Time"] <- r[, "LastSaleTimestamp"]
+  colnames(r)[which(colnames(r) == "LastsaleTimeStamp")] <- "Trade Time"
```
[the remark this answers is not in the document]
Comment (joshuaulrich) [reply R/getQuote.R]: A more robust solution would be to leave L372 as-is, and follow it with `r[, "LastSaleTimestamp"] <- NULL` to remove the `"LastSaleTimestamp"` column.

**Comment by ethanbsmith** [comment]
Good points.

I won’t be able to get to this till after market closes today. If you’re on a roll, I wont be offended if you make the changes


Also, on the Symbols <- unlist(strsplit(Symbols,','))  changes,

I think that processing can be moved up to the master getQuote function. I can do a separate ticket/pr to cover that if you agree on the approach

Not sure how to unit test the av and tiingo functions without a key though

**Comment by ethanbsmith** [comment]
ok, i made some changes. i think the diffs should be much cleaner now. Also, i think it's set up to consolidate the strsplit and merge code into the master function

**Comment by joshuaulrich** [comment]
I think you removed your functional changes, and some code that was in
master (before your PR). I'll fix and merge tomorrow morning, at the latest.

Put your symbol splitting code in another branch, and I'll make sure it
gets merged too.

**Comment by ethanbsmith** [comment]
i realized the `merge` call wasn't needed. you can just use the symbol names returned from the server. If we want to ensure that the output includes NA rows for input symbols, which isn't a bad idea, that can be centralized into the main function, after the getting the dispatched results. If you want that, I can do at the same time as the symbol splitting. I'll do that after this PR gets merged so the conflicts don't get complicated. I already opened an issue to track that

**Comment by joshuaulrich** [comment]
> If we want to ensure that the output includes NA rows for input symbols, which isn't a bad idea, that can be centralized into the main function, after the getting the dispatched results.

We do want that, because that's the existing behavior. Removing it would break backward compatibility. Moving that to the main function should be in a separate PR, so I'm going to revert a71e1e59ed46f95a53187dd2c758637f4dafd26d.

I'm going to leave the `Symbols <- unlist(strsplit(Symbols,';'))` because I'm not sure your updates will work without them.

----

And I just noticed that the `getQuote` functions still error if all the symbols are not valid. You initial description said:

> getQuote would fail if the source did not recognize *any* of the supplied symbols.

(Emphasis mine). I expected these calls to return objects that only contained missing values, but they still error.

```r
quantmod::getQuote(c("FOO", "WYSIWYG"), src = "av", api.key = keyAV)
quantmod::getQuote(c("FOO", "WYSIWYG"), src = "tiingo", api.key = keyTiingo)
quantmod::getQuote(c("FOO", "WYSIWYG"), src = "yahoo")
```

Am I missing something? Maybe I don't understand the initial problem as well as I thought...

**Comment by ethanbsmith** [comment]
the centralized code is already in PR #285. it might simplify this PR if that one gets accepted first. not sure.

> I just noticed that the getQuote functions still error if **all** the symbols are not valid. 

(Emphasis mine). Interesting. I've never encountered that myself and didn't test for it. It's an odd scenario, but probably one we should account for as well. Are you ok if we handle that in this PR or do you want it broken out?

Thx for the info on the keys. i will add test for av and tiingo

**Comment by joshuaulrich** [comment]
> Thx for the info on the keys. i will add test for av and tiingo

To reiterate, please don't push those tests until I'm convinced my keys won't be publicly visible in an error message.

I looked at the issue again, and I see that your example was for `getQuote.yahoo()`.  I confirm that function throws an error if any symbol is not valid.  `getQuote.av()` and `getQuote.tiingo()` add the rows for any missing symbols.  That's why I was confused.  It makes sense now.

I'll take a look at this again after market hours today.

**Comment by ethanbsmith** [comment]
> I just noticed that the getQuote functions still error if all the symbols are not valid.

been stewing on this in the background all day. not as straight forward as I originally thought. it's fairly innocuous in the simple case of 1 or 2 tickers, but what about 2000 or 10000? if every single ticker you passed in is rejected, something is probably wrong. Sort of seems like a different use case from a disagreement on a subset of tickers. Not sure I have a strong view on "correct" behavior either way on this aspect and would support whatever you decide on
